src/utils/PluginDriver.ts

```diff
@@ -360,3 +392,36 @@ export class PluginDriver {
 		}
 	}
 }
+
+// provides uniqueness for unfulfilled hook actions and the dispatch order is useful for debugging
+let actionCount = 0;
+
+function formatAction(pluginName: string, hookName: string, args: Parameters<any>): string {
```

Comment:
One thought here: Currently you are using a set of strings, which means those strings also need to be unique which means you need the actionCount etc.
Also, you are basically formatting the error that does not exist yet, which feels - slightly wrong to me. Especially as it separates the formatting logic from the reporting logic (ok, they are placed next to each other, but logically, there are executed at very different points).
Why not use objects (or tuples `[plugin.name, hookName, args]`) as actions instead? Then you would have even less overhead, but we could have as expensive stuff as we want in the error reporting. And most importantly, the error reporting would be separate.